subsys/bluetooth/mesh/access.c

```diff
@@ -316,6 +317,22 @@ static void mod_init(struct bt_mesh_model *mod, struct bt_mesh_elem *elem,
 	mod->elem_idx = elem - dev_comp->elem;
 	if (vnd) {
 		mod->mod_idx = mod - elem->vnd_models;
+
+		if (IS_ENABLED(CONFIG_BT_MESH_MODEL_VND_MSG_CID_FORCE)) {
+			const struct bt_mesh_model_op *op;
+
+			for (op = mod->op; op->func; op++) {
+				cid = (uint16_t)(op->opcode & 0xffff);
+				if (cid != mod->vnd.company) {
+					*err = -EINVAL;
+					BT_ERR("Invalid vendor model(company:0x%04x"
+					       " id:0x%04x) message opcode 0x%08x",
+						mod->vnd.company, mod->vnd.id, op->opcode);
```

Comment:
Just as a general trick you can use to reduce the need for deep indentation levels: invert the condition and use `continue;`, so the above would become:
```
        if (cid == mod->vnd.company) {
                continue;
        }

        *err = -EINVAL;
        BT_ERR("...");
        return;
```

That might help you avoid splitting message strings etc (though I don't think 8 characters is enough for the above case).

Reply:
...refactoring into helper functions is another way, and in fact deep indentation levels are often (not necessarily in this case) a sign that the function is too complex and should be refactored anyway.

Reply:
Thanks a lot.